superset/utils/core.py

```diff
@@ -1762,3 +1762,29 @@ def parse_boolean_string(bool_str: Optional[str]) -> bool:
         return bool(strtobool(bool_str.lower()))
     except ValueError:
         return False
+
+
+def apply_max_row_limit(max_limit: Optional[int], limit: int) -> int:
```

Comment:
instead of needing to pass in the max row limit, should we access the config value here? Seems like the first argument is always set to the same thing where this function is used

Reply:
You know you're absolutely right, this current sig is really an artifact from earlier iterations of this PR. Will update accordingly 👍